### PR TITLE
standardized YOLOS model card according to template in #36979

### PR DESCRIPTION
**Pull request by EthanV431**
# What does this PR do?

This PR standardizes the YOLOS model card in the docs. I've followed the template in #36979 as best I can, but I'm not sure if I've done it correctly since the information on this model didn't exactly fit 1-to-1 with the template. I got rid of the table and the images since the other PRs for this did that. Let me know if I need to make any changes.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

 
 @stevhliu

### CODE REVIEW

File: docs/source/en/model_doc/yolos.md [hunk 1]
```diff
@@ -13,76 +13,149 @@ specific language governing permissions and limitations under the License.
...
+    <div class="flex flex-wrap space-x-1">
+        <img alt="FlashAttention" src="https://img.shields.io/badge/%E2%9A%A1%EF%B8%8E%20FlashAttention-eae0c8?style=flat">
+    </div>
+    <div class="flex flex-wrap space-x-1">
+        <img alt="SDPA" src="https://img.shields.io/badge/SDPA-DE3412?style=flat&logo=pytorch&logoColor=white">
+    </div>
+</div>
 
-# YOLOS
+## YOLOS
```
Comment (stevhliu) [inline docs/source/en/model_doc/yolos.md]:
```suggestion
# YOLOS
```

File: docs/source/en/model_doc/yolos.md [hunk 2]
```diff
@@ -13,76 +13,149 @@ specific language governing permissions and limitations under the License.
...
 
-<div class="flex flex-wrap space-x-1">
-<img alt="PyTorch" src="https://img.shields.io/badge/PyTorch-DE3412?style=flat&logo=pytorch&logoColor=white">
-<img alt="FlashAttention" src="https://img.shields.io/badge/%E2%9A%A1%EF%B8%8E%20FlashAttention-eae0c8?style=flat">
-<img alt="SDPA" src="https://img.shields.io/badge/SDPA-DE3412?style=flat&logo=pytorch&logoColor=white">
-</div>
+[YOLOS](https://huggingface.co/papers/2106.00666) proposes to just leverage the plain [Vision Transformer (ViT)](vit) for object detection, inspired by DETR. It turns out that a base-sized encoder-only Transformer can also achieve 42 AP on COCO, similar to DETR and much more complex frameworks such as Faster R-CNN.
```
Comment (stevhliu) [inline docs/source/en/model_doc/yolos.md]:
```suggestion
[YOLOS](https://huggingface.co/papers/2106.00666) uses a [Vision Transformer (ViT)](./vit) for object detection with minimal modifications and region priors. It can achieve performance comparable to specialized object detection models and frameworks with knowledge about 2D spatial structures.
```

File: docs/source/en/model_doc/yolos.md [hunk 3]
```diff
@@ -13,76 +13,149 @@ specific language governing permissions and limitations under the License.
...
+
+The model uses 100 learnable object queries and forces the ground-truth boxes to also be exactly 100, filling any extra slots with a special "no object" entry. During training, the Hungarian matching algorithm was used to create optimal one-to-one mappings between each of the queries and each of the annotations. Once each query is matched, the model is updated by applying a standard cross-entropy loss on the predicted classes and a box loss that mixes L1 distance with a generalized IoU term.
```
Comment (stevhliu) [inline docs/source/en/model_doc/yolos.md]: This is probably a bit too in depth to include

```suggestion
```

File: docs/source/en/model_doc/yolos.md [hunk 4]
```diff
@@ -13,76 +13,149 @@ specific language governing permissions and limitations under the License.
...
+```py
+import torch
+from PIL import Image
+import requests
+from transformers import AutoImageProcessor, AutoModelForObjectDetection
 
-<img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/yolos_architecture.png"
-alt="drawing" width="600"/>
+processor = AutoImageProcessor.from_pretrained("hustvl/yolos-base")
+model = AutoModelForObjectDetection.from_pretrained("hustvl/yolos-base").to("cuda")
```
Comment (stevhliu) [inline docs/source/en/model_doc/yolos.md]:
```suggestion
model = AutoModelForObjectDetection.from_pretrained("hustvl/yolos-base", torch_dtype=torch.float16, attn_implementation="sdpa").to("cuda")
```

File: docs/source/en/model_doc/yolos.md [hunk 5]
```diff
@@ -13,76 +13,149 @@ specific language governing permissions and limitations under the License.
...
+for score, label, box in zip(filtered_scores, filtered_labels, pixel_boxes):
+    x0, y0, x1, y1 = box.tolist()
+    print(f"Label {model.config.id2label[label.item()]}: {score:.2f} at [{x0:.0f}, {y0:.0f}, {x1:.0f}, {y1:.0f}]")
 ```
-from transformers import AutoModelForObjectDetection
-model = AutoModelForObjectDetection.from_pretrained("hustvl/yolos-base", attn_implementation="sdpa", torch_dtype=torch.float16)
-...
+
+</hfoption>
+<hfoption id="transformers CLI">
```
Comment (stevhliu) [inline docs/source/en/model_doc/yolos.md]: Passing images is not supported so we can remove the `transformers cli` option

File: docs/source/en/model_doc/yolos.md [hunk 6]
```diff
@@ -13,76 +13,149 @@ specific language governing permissions and limitations under the License.
...
+      --model hustvl/yolos-base \
+      --device 0
 ```
 
-For the best speedups, we recommend loading the model in half-precision (e.g. `torch.float16` or `torch.bfloat16`).
+</hfoption>
+</hfoptions>
 
-On a local benchmark (A100-40GB, PyTorch 2.3.0, OS Ubuntu 22.04) with `float32` and `hustvl/yolos-base` model, we saw the following speedups during inference.
+Quantization reduces the memory burden of large models by representing the weights in a lower precision. Refer to the [Quantization](../quantization/overview) overview for more available quantization backends.
```
Comment (stevhliu) [inline docs/source/en/model_doc/yolos.md]: YOLOS is pretty small so unlikely to benefit from quantization, let's remove this as well

File: docs/source/en/model_doc/yolos.md [hunk 7]
```diff
@@ -13,76 +13,149 @@ specific language governing permissions and limitations under the License.
...
+    outputs = model_int8(**image)
+```
+
+Use the [AttentionMaskVisualizer](https://github.com/huggingface/transformers/blob/beb9b5b02246b9b7ee81ddf938f93f44cfeaad19/src/transformers/utils/attention_visualizer.py#L139) to better understand what tokens the model can and cannot attend to.
 
-If you're interested in submitting a resource to be included here, please feel free to open a Pull Request and we'll review it! The resource should ideally demonstrate something new instead of duplicating an existing resource.
 
-<Tip>
+```py
+from transformers.utils.attention_visualizer import AttentionMaskVisualizer
```
Comment (stevhliu) [inline docs/source/en/model_doc/yolos.md]: Not supported for YOLOS, so we can remove as well

File: docs/source/en/model_doc/yolos.md [hunk 8]
```diff
@@ -13,76 +13,149 @@ specific language governing permissions and limitations under the License.
...
+url   = "https://huggingface.co/datasets/Narsil/image_dummy/raw/main/parrots.png"
+image = Image.open(requests.get(url, stream=True).raw).convert("RGB")
+
+visualizer(image)
+```
 
-</Tip>
+## Notes
+- For the best speedups, we recommend loading the model in half-precision (e.g. `torch.float16` or `torch.bfloat16`).
+- Use [`YolosImageProcessor`] for preparing images (and optional targets) for the model. Contrary to [DETR](detr), YOLOS doesn't require a `pixel_mask` to be created.
```
Comment (stevhliu) [inline docs/source/en/model_doc/yolos.md]:
```suggestion
## Notes
- Use [`YolosImageProcessor`] for preparing images (and optional targets) for the model. Contrary to [DETR](./detr), YOLOS doesn't require a `pixel_mask`.
```

File: docs/source/en/model_doc/yolos.md [hunk 9]
```diff
@@ -13,76 +13,149 @@ specific language governing permissions and limitations under the License.
...
+<hfoption id="Automodel">
 
-*Can Transformer perform 2D object- and region-level recognition from a pure sequence-to-sequence perspective with minimal knowledge about the 2D spatial structure? To answer this question, we present You Only Look at One Sequence (YOLOS), a series of object detection models based on the vanilla Vision Transformer with the fewest possible modifications, region priors, as well as inductive biases of the target task. We find that YOLOS pre-trained on the mid-sized ImageNet-1k dataset only can already achieve quite competitive performance on the challenging COCO object detection benchmark, e.g., YOLOS-Base directly adopted from BERT-Base architecture can obtain 42.0 box AP on COCO val. We also discuss the impacts as well as limitations of current pre-train schemes and model scaling strategies for Transformer in vision through YOLOS.*
+```py
+import torch
+from PIL import Image
+import requests
+from transformers import AutoImageProcessor, AutoModelForObjectDetection
 
-<img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/yolos_architecture.png"
```
Comment (stevhliu) [inline docs/source/en/model_doc/yolos.md:34]: We can keep the image

File: docs/source/en/model_doc/yolos.md [hunk 10]
```diff
@@ -13,76 +13,149 @@ specific language governing permissions and limitations under the License.
...
 
-</Tip>
+## Notes
+- For the best speedups, we recommend loading the model in half-precision (e.g. `torch.float16` or `torch.bfloat16`).
+- Use [`YolosImageProcessor`] for preparing images (and optional targets) for the model. Contrary to [DETR](detr), YOLOS doesn't require a `pixel_mask` to be created.
+
+## Resources
+
+- All example notebooks illustrating inference + fine-tuning [`YolosForObjectDetection`] on a custom dataset can be found [here](https://github.com/NielsRogge/Transformers-Tutorials/tree/master/YOLOS).
+- Scripts for finetuning [`YolosForObjectDetection`] with [`Trainer`] or [Accelerate](https://huggingface.co/docs/accelerate/index) can be found [here](https://github.com/huggingface/transformers/tree/main/examples/pytorch/object-detection).
```
Comment (stevhliu) [inline docs/source/en/model_doc/yolos.md]:
```suggestion
- Refer to these [notebooks](https://github.com/NielsRogge/Transformers-Tutorials/tree/master/YOLOS) for inference and fine-tuning with [`YolosForObjectDetection`] on a custom dataset.
```

**Review by stevhliu — COMMENTED** [review]
Thanks for your contribution!

**Comment by EthanV431** [comment]
messed up the commit tree a bit, but i put in all the suggestions

**Review by stevhliu — APPROVED** [review]
LGTM! 🚀
